### PR TITLE
source-bigquery: escape column names and update google cloud bigquery artifact

### LINKED ISSUE #10383: BigQuery connector doesn't escape column names
## Environment
- **Airbyte version**:  0.29.4-alpha
- **OS Version / Instance**:GCP N2
- **Deployment**: docker
- **Source Connector and version**: BiQuery 0.1.5
- **Destination Connector and version**: N/A
- **Severity**: Medium
- **Step where error happened**: SyncJob

## Current Behavior
If a sync contains a column thats a reserved identifier eg. `interval` we get an exception message from BigQuery stating the sql statement is invalid (it doesn't expect `,` a that point)

## Expected Behavior
Connector should escape the column names when creating queries to BigTable, allowing for column names with reserved identifiers.

## Logs
<details>
<summary>LOG</summary> 

```
2022-02-16 16:59:27 ERROR Exception in thread "main" java.lang.RuntimeException: java.lang.RuntimeException: com.google.cloud.bigquery.BigQueryException: Syntax error: Unexpected "," at [1:879]
2022-02-16 16:59:27 ERROR 	at io.airbyte.integrations.source.relationaldb.AbstractRelationalDbSource.lambda$queryTable$0(AbstractRelationalDbSource.java:64)
2022-02-16 16:59:27 ERROR 	at io.airbyte.commons.util.LazyAutoCloseableIterator.computeNext(LazyAutoCloseableIterator.java:37)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:146)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:141)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.TransformedIterator.hasNext(TransformedIterator.java:46)
2022-02-16 16:59:27 ERROR 	at io.airbyte.commons.util.DefaultAutoCloseableIterator.computeNext(DefaultAutoCloseableIterator.java:38)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:146)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:141)
2022-02-16 16:59:27 ERROR 	at io.airbyte.integrations.source.relationaldb.StateDecoratingIterator.computeNext(StateDecoratingIterator.java:48)
2022-02-16 16:59:27 ERROR 	at io.airbyte.integrations.source.relationaldb.StateDecoratingIterator.computeNext(StateDecoratingIterator.java:18)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:146)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:141)
2022-02-16 16:59:27 ERROR 	at io.airbyte.commons.util.DefaultAutoCloseableIterator.computeNext(DefaultAutoCloseableIterator.java:38)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:146)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:141)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.TransformedIterator.hasNext(TransformedIterator.java:46)
2022-02-16 16:59:27 ERROR 	at io.airbyte.commons.util.DefaultAutoCloseableIterator.computeNext(DefaultAutoCloseableIterator.java:38)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:146)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:141)
2022-02-16 16:59:27 ERROR 	at io.airbyte.commons.util.CompositeIterator.computeNext(CompositeIterator.java:63)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:146)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:141)
2022-02-16 16:59:27 ERROR 	at io.airbyte.commons.util.DefaultAutoCloseableIterator.computeNext(DefaultAutoCloseableIterator.java:38)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:146)
2022-02-16 16:59:27 ERROR 	at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:141)
2022-02-16 16:59:27 ERROR 	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:132)
2022-02-16 16:59:27 ERROR 	at io.airbyte.integrations.base.IntegrationRunner.run(IntegrationRunner.java:119)
2022-02-16 16:59:27 ERROR 	at io.airbyte.integrations.source.bigquery.BigQuerySource.main(BigQuerySource.java:175)
2022-02-16 16:59:27 ERROR Caused by: java.lang.RuntimeException: com.google.cloud.bigquery.BigQueryException: Syntax error: Unexpected "," at [1:879]
2022-02-16 16:59:27 ERROR 	at io.airbyte.db.bigquery.BigQueryDatabase.waitForQuery(BigQueryDatabase.java:201)
2022-02-16 16:59:27 ERROR 	at io.airbyte.db.bigquery.BigQueryDatabase.executeQuery(BigQueryDatabase.java:185)
2022-02-16 16:59:27 ERROR 	at io.airbyte.db.bigquery.BigQueryDatabase.executeQuery(BigQueryDatabase.java:135)
2022-02-16 16:59:27 ERROR 	at io.airbyte.db.bigquery.BigQueryDatabase.query(BigQueryDatabase.java:106)
2022-02-16 16:59:27 ERROR 	at io.airbyte.db.bigquery.BigQueryDatabase.query(BigQueryDatabase.java:102)
2022-02-16 16:59:27 ERROR 	at io.airbyte.integrations.source.relationaldb.AbstractRelationalDbSource.lambda$queryTable$0(AbstractRelationalDbSource.java:61)
2022-02-16 16:59:27 ERROR 	... 27 more
2022-02-16 16:59:27 ERROR Caused by: com.google.cloud.bigquery.BigQueryException: Syntax error: Unexpected "," at [1:879]
2022-02-16 16:59:27 ERROR 	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.translate(HttpBigQueryRpc.java:115)
2022-02-16 16:59:27 ERROR 	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.getQueryResults(HttpBigQueryRpc.java:628)
2022-02-16 16:59:27 ERROR 	at com.google.cloud.bigquery.BigQueryImpl$35.call(BigQueryImpl.java:1347)
2022-02-16 16:59:27 ERROR 	at com.google.cloud.bigquery.BigQueryImpl$35.call(BigQueryImpl.java:1342)
2022-02-16 16:59:27 ERROR 	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:105)
2022-02-16 16:59:27 ERROR 	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
2022-02-16 16:59:27 ERROR 	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
2022-02-16 16:59:27 ERROR 	at com.google.cloud.bigquery.BigQueryImpl.getQueryResults(BigQueryImpl.java:1341)
2022-02-16 16:59:27 ERROR 	at com.google.cloud.bigquery.BigQueryImpl.getQueryResults(BigQueryImpl.java:1325)
2022-02-16 16:59:27 ERROR 	at com.google.cloud.bigquery.Job$1.call(Job.java:334)
2022-02-16 16:59:27 ERROR 	at com.google.cloud.bigquery.Job$1.call(Job.java:331)
2022-02-16 16:59:27 ERROR 	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:105)
2022-02-16 16:59:27 ERROR 	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
2022-02-16 16:59:27 ERROR 	at com.google.cloud.RetryHelper.poll(RetryHelper.java:64)
2022-02-16 16:59:27 ERROR 	at com.google.cloud.bigquery.Job.waitForQueryResults(Job.java:330)
2022-02-16 16:59:27 ERROR 	at com.google.cloud.bigquery.Job.waitFor(Job.java:240)
2022-02-16 16:59:27 ERROR 	at io.airbyte.db.bigquery.BigQueryDatabase.waitForQuery(BigQueryDatabase.java:199)
2022-02-16 16:59:27 ERROR 	... 32 more
```

</details>

## Steps to Reproduce
1. Create a BQ table containing `interval` as a column
2. Create a connection with above BQ table as a source, pushing to any destination
3. Observe sync job is failing

## Are you willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by mkhokh-33**
## What
Escape column tames for big query source.
If a sync contains a column thats a reserved identifier eg. interval we get an exception message from BigQuery stating the sql statement is invalid 

## How
Escape column tames for big query source during selection with spec char '`'

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-bigquery/src/test-integration/java/io/airbyte/integrations/source/bigquery/BigQuerySourceEscapeColumnNameTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import java.sql.SQLException;
+import java.util.List;
+
+import static io.airbyte.integrations.source.bigquery.BigQuerySource.CONFIG_CREDS;
+import static io.airbyte.integrations.source.bigquery.BigQuerySource.CONFIG_DATASET_ID;
+import static io.airbyte.integrations.source.bigquery.BigQuerySource.CONFIG_PROJECT_ID;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class BigQuerySourceEscapeColumnNameTest {
```
Comment (DoNotPanicUA) [inline airbyte-integrations/connectors/source-bigquery/src/test-integration/java/io/airbyte/integrations/source/bigquery/BigQuerySourceEscapeColumnNameTest.java:32]: As this test is copy of the `BigQuerySourceTest` test, please avoid copypasting. We can merge the or create an abstract class for such tests.

**Review by DoNotPanicUA — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-bigquery/src/main/java/io/airbyte/integrations/source/bigquery/BigQuerySource.java [hunk 2]
```diff
@@ -42,7 +42,7 @@
   public static final String CONFIG_PROJECT_ID = "project_id";
   public static final String CONFIG_CREDS = "credentials_json";
 
-  private final String quote = "";
+  private final String quote = "`";
```
Comment (etsybaev) [inline airbyte-integrations/connectors/source-bigquery/src/main/java/io/airbyte/integrations/source/bigquery/BigQuerySource.java:45]: static final String QUOTE?

**Review by etsybaev — APPROVED** [review]
Left a minor comment. Approving in advance, PR to fix as per comments. Thanks

**Review by DoNotPanicUA — APPROVED** [review]

**Review by edgao — APPROVED** [review]

**Pull request merged**